src/test/ui/issues/issue-17551.stderr

```diff
@@ -1,8 +1,8 @@
-error[E0282]: type annotations needed
+error[E0282]: type annotations needed in `B<T>`
   --> $DIR/issue-17551.rs:6:15
    |
 LL |     let foo = B(marker::PhantomData);
-   |         ---   ^ cannot infer type for `T`
+   |         ---   ^ cannot infer type for `T` in `B<T>`
```

Comment:
I cannot tell that `T` is a type variable from this output.

Reply:
What do you mean?

Would you prefer:

```
error[E0282]: type annotations needed for `std::vec::Vec<_>`
  --> $DIR/vector-no-ann.rs:2:16
   |
LL |     let foo = Vec::new();
   |         ---   ^^^^^^^^ cannot infer type for `T`
   |         |
   |         consider giving `foo` the type `std::vec::Vec<T>` with the type parameter `T` specified
```

Reply:
Yeah; tho suggesting turbofish might be better and less invasive?

Reply:
I think we bias our suggestions to first suggest giving a type to `let` bindings, and only if that is not possible suggest the turbofish. I would rather we continued to follow that unofficial guideline.